### PR TITLE
JSONL Support Added

### PR DESCRIPTION
**Pull request by fahadsiddiqui**
## Changes

1. Added JSON lines support https://jsonlines.org/examples/

## Checklist

Checklist for for pull requests including new code and/or changes to existing code...

* [ ] Code
  * [x] Includes unit tests
  * [x] New functions have docstrings with examples that can be run with doctest
  * [ ] New functions are included in API docs
  * [ ] Docstrings include notes for any changes to API or behaviour
  * [x] All changes documented in docs/changes.rst
* [ ] Testing
  * [ ] \(Optional) Tested local against remote servers
  * [x] Travis CI passes (unit tests run under Linux)
  * [x] AppVeyor CI passes (unit tests run under Windows)
  * [ ] Unit test coverage has not decreased (see Coveralls)
* [ ] Changes
  * [ ] \(Optional) Just a proof of concept
  * [ ] \(Optional) Work in progress
  * [x] Ready to review
  * [ ] Ready to merge

### CODE REVIEW

**Comment by fahadsiddiqui** [comment]
@juarezr sorry, I closed last PR due to some checks failing. Made this after fixing those and adding some test cases too.

**Review by alimanfoo — APPROVED** [review]
This looks great to me, just needs an entry in docs/changes.rst.

**Comment by fahadsiddiqui** [comment]
@alimanfoo I made the changes, would it automatically increment version to 1.6.7 when this will be published? Please review.

**Comment by fahadsiddiqui** [comment]
@juarezr can you merge this?

**Comment by juarezr** [comment]
Hi @fahadsiddiqui 

Current version is [v1.6.5](https://github.com/petl-developers/petl/releases).
Next version will be `v1.6.6`. There is also some others commits pending release.

Could you append to `v1.6.6` in `changes.rst`, please?

Do you need a release with this feature soon ?

File: docs/changes.rst [hunk 1]
```diff
@@ -1,6 +1,13 @@
 Changes
 =======
 
+Version 1.6.7
```
Comment (juarezr) [inline docs/changes.rst:4]: Next version is [1.6.6](https://github.com/petl-developers/petl/releases)

**Review by juarezr — CHANGES_REQUESTED** [review]

File: docs/changes.rst (hunk 1 above)
Comment (fahadsiddiqui) [reply docs/changes.rst]: I have made the changes. And yes I will need the feature soon, my time is using PETL heavily.

**Review by juarezr — APPROVED** [review]

**Comment by juarezr** [comment]
Released in [v1.6.6](https://github.com/petl-developers/petl/releases/tag/v1.6.6).

Packages for `pip` and `conda`will be available soon.

**Comment by juarezr** [comment]
Added a reference in wardi/jsonlines#41 documentation.

**Pull request merged**